### PR TITLE
fix bootstrapping bug and add real test for bootstrapping

### PR DESCRIPTION
**Pull request by whyrusleeping**
License: MIT
Signed-off-by: Jeromy <jeromyj@gmail.com>

### CODE REVIEW

File: core/bootstrap.go [hunk 1]
```diff
@@ -219,20 +219,7 @@ func toPeerInfos(bpeers []config.BootstrapPeer) []peer.PeerInfo {
...
 
-	return peers
-}
-
-func toPeerInfo(bp config.BootstrapPeer) peer.PeerInfo {
-	// for now, we drop the "ipfs addr" part of the multiaddr. the rest
-	// of the codebase currently uses addresses without the peerid part.
-	m := bp.Multiaddr()
-	s := ma.Split(m)
-	m = ma.Join(s[:len(s)-1]...)
```
Comment (whyrusleeping) [inline core/bootstrap.go]: this whole bit could have just been 'bp.Transport()'

File: test/sharness/t0121-bootstrap-iptb.sh [hunk 2]
```diff
@@ -0,0 +1,65 @@
+#!/bin/sh
+#
+# Copyright (c) 2016 Jeromy Johnson
```
Comment (whyrusleeping) [inline test/sharness/t0121-bootstrap-iptb.sh]: i stopped and admired the fact that the script i copied this from was 2014. I had to bump it two years forward

File: core/bootstrap.go [hunk 3]
```diff
@@ -202,7 +201,7 @@ func bootstrapConnect(ctx context.Context, ph host.Host, peers []peer.PeerInfo)
 	return nil
 }
 
-func toPeerInfos(bpeers []config.BootstrapPeer) []peer.PeerInfo {
+func toPeerInfos(bpeers []config.BootstrapPeer) ([]peer.PeerInfo, error) {
```
Comment (jbenet) [inline core/bootstrap.go]: where do you make use of the error?

Comment (jbenet) [inline core/bootstrap.go]: i only see `return peers, nil`

File: test/sharness/t0121-bootstrap-iptb.sh [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+check_has_connection() {
+	node=$1
+	ipfsi $node swarm peers | grep ipfs > /dev/null
+}
+
+test_expect_success "setup iptb nodes" '
+	iptb init -n 5 -f --bootstrap=none --port=0
+'
+
```
Comment (jbenet) [inline test/sharness/t0121-bootstrap-iptb.sh]: let's try this here:

```sh
test_expect_success "start up iptb nodes" '
    iptb start --wait
'

test_expect_success "check peers works" '
    ipfs swarm peers >peers_out
'

test_expect_success "correct number of peers" '
    test_empty $(cat peers_out)
'
```

File: core/bootstrap.go (hunk 3 above)
Comment (whyrusleeping) [inline core/bootstrap.go]: oh whoops... I was using the `iaddr.ParseMultiaddr` to parse the `/ipfs/Qmasdasd` part off of the addresses, then i realized that was already being done for me.

File: test/sharness/t0121-bootstrap-iptb.sh (hunk 4 above)
Comment (whyrusleeping) [inline test/sharness/t0121-bootstrap-iptb.sh]: why is that better?

File: test/sharness/t0121-bootstrap-iptb.sh [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+test_expect_success "start up iptb nodes" '
+	iptb start --wait
+'
+
+test_expect_success "check peers works" '
+	ipfs swarm peers > peers_out
+'
+
+test_expect_success "correct number of peers" '
+	test `cat peers_out | wc -l` == 5
```
Comment (jbenet) [inline test/sharness/t0121-bootstrap-iptb.sh]: - https://travis-ci.org/ipfs/go-ipfs/jobs/102864393#L6628
- i think you want `== "5"` ?

File: core/bootstrap.go (hunk 3 above)
Comment (jbenet) [inline core/bootstrap.go]: also https://travis-ci.org/ipfs/go-ipfs/jobs/102864391#L183

File: test/sharness/t0121-bootstrap-iptb.sh (hunk 2 above)
Comment (jbenet) [inline test/sharness/t0121-bootstrap-iptb.sh]: :+1:

**Comment by lidel** [comment]
Thank you. I retested my setup and bootstrap works fine now :+1:

**Comment by whyrusleeping** [comment]
@lidel yay!!

**Pull request merged**